app/src/main/java/fr/free/nrw/commons/upload/MultipleShareActivity.java

```diff
@@ -275,20 +316,60 @@ protected void onAuthCookieAcquired(String authCookie) {
         isMultipleUploadsPrepared = false;
         mwApi.setAuthCookie(authCookie);
         if (!ExternalStorageUtils.isStoragePermissionGranted(this)) {
-            ExternalStorageUtils.requestExternalStoragePermission(this);
+            //If permission is not there, handle the negative cases
+            askDexterToHandleExternalStoragePermission();
             isMultipleUploadsPrepared = false;
             return; // Postpone operation to do after gettion permission
         } else {
             isMultipleUploadsPrepared = true;
-            prepareMultipleUpoadList();
+            prepareMultipleUploadList();
+        }
+    }
+
+    private void askDexterToHandleExternalStoragePermission() {
+        Timber.d(TAG, "External storage permission is being requested");
+        if (null == dexterStoragePermissionBuilder) {
+            dexterStoragePermissionBuilder = Dexter.withActivity(this)
+                    .withPermission(permission.WRITE_EXTERNAL_STORAGE)
+                    .withListener(new BasePermissionListener() {
+                        @Override
+                        public void onPermissionGranted(PermissionGrantedResponse response) {
+                            Timber.d(TAG,
+                                    "User has granted us the permission for writing the external storage");
```

Comment:
Please don't split this line, it does not improve readability. :)

Reply:
Will do